mobile/lib/pages/editing/edit.page.dart

```diff
@@ -0,0 +1,107 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:fluttertoast/fluttertoast.dart';
+import 'package:immich_mobile/pages/editing/crop.page.dart';
+import 'package:immich_mobile/entities/asset.entity.dart';
+import 'package:immich_mobile/widgets/common/immich_image.dart';
+import 'package:immich_mobile/widgets/common/immich_toast.dart';
+@immutable
+class EditImagePage extends StatelessWidget {
+  final Asset? asset;
+  final Image? image;
+  const EditImagePage({
+    super.key,
+    this.image,
+    this.asset,
+  }) : assert((image != null && asset == null) || (image == null && asset != null), 'Must supply one of asset or image');
+  ImageProvider _getImageProvider() {
+    if (asset is Asset) {
+      return ImmichImage.imageProvider(asset: asset);
+    } else if (image is Image) {
+      return (image as Image).image;
```

Comment:
Not necessary to cast `image as Image`

Reply:
Removed the whole function and placed it inline.